ops/framework.py

```diff
@@ -489,6 +489,13 @@ def __repr__(self):
         event_kinds = ', '.join(sorted(self._event_kinds()))
         return f'<{k.__module__}.{k.__qualname__}: {event_kinds}>'
 
+    def __getattr__(self, name: str):
+        """This tells type checkers to suspend judgement on this type's attributes.
+
+        This allows charms to freely write `self.on.foobar_baz` without mypy/pyright whining.
```

Comment:
Maybe we can give a more realistic example in the docstring, like "This allows charms to access dynamically-defined events such as `self.on.db_relation_joined` without Mypy/Pyright whining."